Drivers/Comms/Src/Xbee.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Xbee.cpp
+ *
+ *  Created on: Aug 20, 2022
+ *      Author: Aaditya
+ */
+
+#include "Xbee.hpp"
+#include "main.h"
+#include "usart.h"
+
+#define UART_TIMEOUT 100
+
+#define DRONE_BASIC_PAYLOAD_TYPE 1
+
+void changeEndian(uint8_t* val, int size) {
+	uint8_t temp;
+	for (int x = 0; x < size/2; x++) {
+		temp = val[x];
+		val[x] = val[size - 1 - x];
+		val[size - 1 - x] = temp;
+	}
+}
+
+void TransmitMessage(DroneBasicPayload* payload) {
+	Packet packet = {};
+	TransmitFrameHeader tf = {};
+
+	packet.time = HAL_GetTick();
+	packet.payloadType = DRONE_BASIC_PAYLOAD_TYPE;
+
+	uint8_t checksum = 0xFF;
+
+	uint8_t rawPacket[sizeof(TransmitFrameHeader) + sizeof(Packet) + sizeof(DroneBasicPayload) + sizeof(checksum)];
+
+
+	uint8_t* tfArr = (uint8_t*)&tf;
+	uint8_t* packetArr = (uint8_t*)&packet;
+	uint8_t* payloadArr = (uint8_t*)payload;
+
+	tf.messageLength = sizeof(TransmitFrameHeader) + sizeof(Packet) + sizeof(DroneBasicPayload) - 3;
+
+	tf.shortAddress = (tf.shortAddress & 0xFF00) >> 8 | (tf.shortAddress & 0x00FF) << 8;
```

Comment:
Hold up, for line 43, `tf.shortAddress = (tf.shortAddress & 0xFF00) >> 8 | (tf.shortAddress & 0x00FF) << 8` are you switching the endianness? Bec you are only switching the first and last byte, slightly confused here.

Reply:
Im switching the endianness. Its a 16 bit uint, so theres only 2 bytes. I could do it with the function, but idk this way seems to work fine as well. Considering it's only 2 bytes I think its fine.

Reply:
I would prefer if it were in a function, somewhere common, but if this works then this works.